the-collab-lab/tcl-25-smart-shopping-list
# [WIP] Issue #3 | Generate a token for setting up a shopping list

## Description


This PR allows users to set up a shopping list so that they can start tracking purchased items.  The PR makes sure that users with a token already stored in localStorage are shown the `list` view. Those without a token can generate one by clicking the `Create a Shopping List` button. Once the token is generated, it gets stored  into the browser localStorage, and users are shown the `list` view .


## Acceptance Criteria

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with `git pull origin ku-cz-set-up-shopping-list` and check that branch out with `git checkout ku-cz-set-up-shopping-list` 
* Then `npm start` to launch the app
* Click the `Create a Shopping List` button to generate a token. Make sure generating the token redirects you to the `List` view
* To see the token in `localStorage`, open developer tools and click on `Application` tab. Find the token under the `localStorage` dropdown menu on left side of the screen
* If you have a token already saved in `localStorage`, `ctrl c` to stop the app, `npm start` to launch it again. Make sure you get shown the `List` view
